src/uart.rs

```diff
@@ -428,12 +573,61 @@ crate::embedded_hal_error!(
     embedded_hal_nb::serial::ErrorKind
 );
 
+#[derive(Clone, Copy)]
+#[repr(transparent)]
+pub struct UartEvent {
+    raw: uart_event_t,
+}
+
+impl UartEvent {
+    pub fn payload(&self) -> UartEventPayload {
+        #[allow(non_upper_case_globals)]
+        match self.raw.type_ {
+            uart_event_type_t_UART_DATA if self.raw.timeout_flag => UartEventPayload::Timeout,
+            uart_event_type_t_UART_DATA => UartEventPayload::Data {
+                size: self.raw.size,
+            },
+            uart_event_type_t_UART_BREAK => UartEventPayload::Break,
+            uart_event_type_t_UART_BUFFER_FULL => UartEventPayload::RxBufferFull,
+            uart_event_type_t_UART_FIFO_OVF => UartEventPayload::RxFifoOverflow,
+            uart_event_type_t_UART_FRAME_ERR => UartEventPayload::FrameError,
+            uart_event_type_t_UART_PARITY_ERR => UartEventPayload::ParityError,
+            uart_event_type_t_UART_DATA_BREAK => UartEventPayload::DataBreak,
+            uart_event_type_t_UART_PATTERN_DET => UartEventPayload::PatternDetected,
+            _ => UartEventPayload::Unknown,
+        }
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+#[non_exhaustive]
+pub enum UartEventPayload {
+    /// UART data was received
+    Data {
+        size: usize,
+    },
+    /// Represents DATA event with timeout_flag set
+    Timeout,
```

Comment:
Isn't `size` also relevant for the `Timeout` case? Not sure, just asking.

Reply:
If a byte was received, it wouldn't timeout, would it?

Reply:
No idea. I bet you know the driver better than me at this point. Just asking if this was a typing omission or suchlike.

Reply:
It could happen that both interrupt flags were set in case the timeout couldn't be processed before new data arrived. It seems weird that we don't get two events in that case, so maybe considering that, it makes more sense to have all of that in a single variant... I will document it accordingly.